src/NServiceBus.Core/Transports/IMessageDispatcher.cs

```diff
@@ -10,6 +11,6 @@ public interface IMessageDispatcher
         /// <summary>
         /// Dispatches the given operations to the transport.
         /// </summary>
-        Task Dispatch(TransportOperations outgoingMessages, TransportTransaction transaction);
+        Task Dispatch(TransportOperations outgoingMessages, TransportTransaction transaction, CancellationToken cancellationToken);
```

Comment:
If I'm not mistaken the dispatcher is registered on DI and for advanced raw dispatch scenarios, it is possible to inject yourself a dispatcher (we do that as well in several downstream for raw messaging scenarios including ServiceControl acceptance tests). Maybe it makes sense to make the token here optional?

Reply:
I'd say no given that this is an advanced API breaking users isn't a big concern since the fix is obvious?

Reply:
My thought was that in general, we made cancellation optional for all our APIs that are exposed to the user. The fact that this class can be injected and used as an API to send raw messages made me think if the same rules should apply here like for the other user exposed APIs. I get what you are saying for the other internal APIs that there it should be mandatory because only core ever will call those. The dispatcher can be called though by any user (even though advanced but we kind of applied the same optional principle to the behavior pipeline by putting the token on the context)

Reply:
Discussed with @danielmarbach and we agreed to consider this API a downstream API even though users might be able to access it and there for following the stricter standard with a mandatory token